olp-cpp-sdk-core/src/http/curl/NetworkCurl.cpp

```diff
@@ -176,6 +176,30 @@ static curl_code SslctxFunction(CURL* curl, void* sslctx, void*) {
 #endif
 #endif
 
+/**
+ * @brief CURL get upload/download data.
+ * @param[in] handle CURL easy handle.
+ * @param[out] upload_bytes uploaded bytes(headers+data).
+ * @param[out] download_bytes downloaded bytes(headers+data).
+ */
+void GetTraficData(CURL* handle, uint64_t& upload_bytes,
+                   uint64_t& download_bytes) {
+  long headers_size;
+  double length_downloaded;
+  if (curl_easy_getinfo(handle, CURLINFO_HEADER_SIZE, &headers_size) ==
+          CURLE_OK &&
+      curl_easy_getinfo(handle, CURLINFO_SIZE_DOWNLOAD, &length_downloaded) ==
+          CURLE_OK) {
+    download_bytes = length_downloaded + headers_size;
+  }
```

Comment:
I would rather separate it, currently, if the first call failed, the second one is not called.

Reply:
it was done intentionally. Why we should count any data, if some fail happen?